server/handler/resources.go

```diff
@@ -0,0 +1,69 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/kubernetes-sigs/kube-scheduler-simulator/resources"
+	"github.com/kubernetes-sigs/kube-scheduler-simulator/server/di"
+	"github.com/labstack/echo/v4"
+	"k8s.io/klog/v2"
+)
+
+type ResourcesHandler struct {
+	service di.ResourcesService
+}
+
+func NewResourcesHandler(s di.ResourcesService) *ResourcesHandler {
+	return &ResourcesHandler{service: s}
+}
+
+func (h *ResourcesHandler) ExportResourcesAll(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	rs, err := h.service.ExportAll(ctx)
+	if err != nil {
+		klog.Errorf("failed to export all resources: %+v", err)
+		return echo.NewHTTPError(http.StatusInternalServerError)
+	}
+	return c.JSON(http.StatusOK, rs)
+}
+
+func (h *ResourcesHandler) ImportResourcesAll(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	// backup before overwrite exist resources.
+	bkp, err := h.service.ExportAll(ctx)
```

Comment:
I think it is better to assume that this import feature can only be used when a user has not created any resources. So, it's not needed to think about existing resources, not needed to take backup.

Reply:
Import is processed in order for each service.
So what should I do if it fails in the middle of the Apply sequence? There are some resources that have already been imported and made. (If there is a way to check if all the imported data is correct before starting Apply...)

Reply:
> So what should I do if it fails in the middle of the Apply sequence?

I think you don't need to do anything. This will simplify the behavior of the API.
If users want to delete resources that have already been imported, they can use reset resources with reset button. #26